### ISSUE TITLE
Clean up mirror repos

### ISSUE THREAD

**Issue by mattwynne** [cucumber/common#1550]
When we first moved to the monorepo, many of the package managers we used needed a library to be in it's own repo, so we set up individual read-only mirror repos which are copied to when there are changes in this repo.

The context has changed, and we don't need most of these any more. We might still need them for the go modules.

**Comment by mattwynne** [comment]
@jaysonesmith or anyone else from @cucumber/go can you help me understand whether we still need the subrepos for publishing common go modules?

**Comment by aurelien-reeves** [comment]
We can easily specify a go module which is in a monorepo:
https://golang.org/ref/mod#module-path
https://golang.org/ref/mod#vcs-dir
So in our case, that should be as simple as something like the following: github.com/cucumber/common/messages/go/v42

**Comment by ciaranmcnulty** [comment]
Would be great if this eliminates the rsync'ing

**Comment by aslakhellesoy** [comment]
I disagree with `use symlinks instead of rsync and delete rsync scripts/functions` - symlinks work poorly in git.

We may be able to replace some of the copied files with *relative file paths* within the repo, but we might not be able to do it for all rsync'ed files.

For example, I think `.mocharc.json` files must be in the same directory as `package.json` - there might be more cases like that.

**Comment by ciaranmcnulty** [comment]
> symlinks work poorly in git

Is that a Windows issue?

**Comment by aslakhellesoy** [comment]
I thought it was a Windows issue (used to be at least), but have no simple way of verifying if it works now. Maybe it does now: https://github.community/t/windows-symlinks-do-they-play-well-with-git-projects/730

If symlinks work on Windows too, it would certainly simplify things!

**Comment by ciaranmcnulty** [comment]
I have similar vague memories :)

I think under WSL2 it's fine but no idea how it works on 'real' Windows nowadays

**Comment by ciaranmcnulty** [comment]
If we don't have separate repos in future though I agree that we should minimise the use of symlinks in favour of relative paths

**Comment by mattwynne** [comment]
@gasparnagy @aurelien-reeves @enkessler can you add anything about the symlinks situation on Windows these days?

**Comment by gasparnagy** [comment]
@mattwynne did a quick check. you can create symlinks, but only with admin privileges. so with normal console where you would do a git pull it does not work.

How does git support symlinks. Could anyone point me to a git repo that had symlinks inside - i would try to see what happens if i pull...

btw the win command for creating symlinks is

```
mklink symlink-file target-file
```

**Comment by gasparnagy** [comment]
It seems you can enable creating symlinks without admin, but it requires you to enable developer mode (i don't have that enabled... and never heard about it yet). Unfortunately developer mode enables a bunch of other things, like SSH server, that you might not want to have on a laptop for example...
https://docs.microsoft.com/en-us/windows/apps/get-started/enable-your-device-for-development

**Comment by mattwynne** [comment]
@gasparnagy try https://github.com/mattwynne/symlink-test ?

**Comment by mattwynne** [comment]
You should be able to do:

```
> type link-to-readme.md
# symlink-test
```

**Comment by aurelien-reeves** [comment]
> have separate repos in future though I agree that we should minimise the use of symlinks i

I agree.
Why to use symlinks where relative path may be enough?

Symlinks may still be confusing sometimes where relative path are explicit anyway.

**Comment by gasparnagy** [comment]
> @gasparnagy try https://github.com/mattwynne/symlink-test ?

Does not work. Git clone just clones it as normal file that contains the name of the linked file... :(

```
> type link-to-readme.md
README.md
```

So I think we can say that we should not use symlinks if we want to have contributors with Windows...

**Comment by mattwynne** [comment]
> Why to use symlinks where relative path may be enough?

Of course! I hadn't thought of that. Relative paths should work just fine.
